assert/assertions_test.go

```diff
@@ -544,6 +548,70 @@ func TestNotEqual(t *testing.T) {
 	if NotEqual(mockT, &struct{}{}, &struct{}{}) {
 		t.Error("NotEqual should return false")
 	}
+
+	// A case that might be confusing, especially with numeric literals
+	if !NotEqual(mockT, 10, uint(10)) {
+		t.Error("NotEqual should return false")
+	}
+}
+
+func TestNotEqualValues(t *testing.T) {
+
+	mockT := new(testing.T)
+
+	// Same tests as NotEqual since they behave the same when types are irrelevant
+	if !NotEqualValues(mockT, "Hello World", "Hello World!") {
+		t.Error("NotEqualValues should return true")
+	}
+	if !NotEqualValues(mockT, 123, 1234) {
+		t.Error("NotEqualValues should return true")
+	}
+	if !NotEqualValues(mockT, 123.5, 123.55) {
+		t.Error("NotEqualValues should return true")
+	}
+	if !NotEqualValues(mockT, []byte("Hello World"), []byte("Hello World!")) {
+		t.Error("NotEqualValues should return true")
+	}
+	if !NotEqualValues(mockT, nil, new(AssertionTesterConformingObject)) {
+		t.Error("NotEqualValues should return true")
+	}
+	if NotEqualValues(mockT, nil, nil) {
+		t.Error("NotEqualValues should return false")
+	}
+	if NotEqualValues(mockT, "Hello World", "Hello World") {
+		t.Error("NotEqualValues should return false")
+	}
+	if NotEqualValues(mockT, 123, 123) {
+		t.Error("NotEqualValues should return false")
+	}
+	if NotEqualValues(mockT, 123.5, 123.5) {
+		t.Error("NotEqualValues should return false")
+	}
+	if NotEqualValues(mockT, []byte("Hello World"), []byte("Hello World")) {
+		t.Error("NotEqualValues should return false")
+	}
+	if NotEqualValues(mockT, new(AssertionTesterConformingObject), new(AssertionTesterConformingObject)) {
+		t.Error("NotEqualValues should return false")
+	}
+	if NotEqualValues(mockT, &struct{}{}, &struct{}{}) {
+		t.Error("NotEqualValues should return false")
+	}
+
+	// Special cases where NotEqualValues behaves differently
+	funcA := func() int { return 23 }
+	funcB := func() int { return 42 }
+	if !NotEqualValues(mockT, funcA, funcB) {
+		t.Error("NotEqualValues should return true")
+	}
+	if !NotEqualValues(mockT, int(10), int(11)) {
+		t.Error("NotEqualValues should return true")
+	}
+	if NotEqualValues(mockT, int(10), uint(10)) {
+		t.Error("NotEqualValues should return false")
+	}
+	if NotEqualValues(mockT, struct{}{}, struct{}{}) {
+		t.Error("NotEqualValues should return false")
+	}
```

Comment:
This is not a blocker, but would these tests be neater if we converted them to table-driven tests? (e.g. https://dave.cheney.net/2019/05/07/prefer-table-driven-tests)

Happy to leave it as your call if you want to change them.

Reply:
I agree - I just copied the testing patterns used elsewhere (and partially just copied the `NotEqual` tests), and I tend to overengineer things :)

And if I would improve it here, why not improve other tests as well, etc.

But I'll give it a shot - expect an update later today.

Reply:
If you would like to have a shot at converting other cases to table-driven tests, we'd greatly appreciate the time and effort. Just a key point, as you mentioned as well, is to keep them in separate PRs. It makes the review process much easier and quick.

If you would like, I'd be happy to merge this PR and you can refactor the tests in separate PRs?

Reply:
Let's do that then. And I will try to refactor at least some tests in a separate PR